File: src/main/java/de/fhswf/genericapplication/models/core/PDAttributeMeta.java
package de.fhswf.genericapplication.models.core;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.xml.bind.annotation.*;
import java.util.ArrayList;
import java.util.List;

@XmlRootElement(namespace = "core", name = "attribute")
@XmlAccessorType(XmlAccessType.FIELD)
public class PDAttributeMeta extends PDMemberMeta {
    @XmlAttribute
    @JsonProperty
    private String dataType = "String";

    @XmlAttribute
    @JsonProperty
    private boolean required = false;

    @XmlElementWrapper(namespace = "core", name = "validators")
    @XmlElement(namespace = "core", name = "validator")
    @JsonProperty
    private List<ValidatorMeta> validators = new ArrayList<>();

    @XmlAttribute
    @JsonProperty
    private String values;
}

File: src/main/ui/src/models/core/PDAttributeMeta.ts
import type {PDMemberMeta} from "@/models/core/PDMemberMeta";
import type {PDValidatorMeta} from "@/models/core/PDValidatorMeta";
import type {PDEnumMeta} from "@/models/core/PDEnumMeta";
import type {DataTypeET} from "@/models/core/DataTypeET";

export interface PDAttributeMeta extends PDMemberMeta {
    required: boolean;
    dataType: DataTypeET;
    validators?: PDValidatorMeta[];
    values?: PDEnumMeta;
}

export function isPDAttributeMeta(obj: any): obj is PDAttributeMeta {
    return 'required' in obj && 'dataType' in obj;
}
